pip_audit/_cli.py

```diff
@@ -317,13 +323,12 @@ def audit() -> None:
                 args.require_hashes,
                 state,
             )
-        elif args.pyproject is not None:
-            pyproject_file = Path(args.pyproject.name)
-            source = PyProjectSource(
-                pyproject_file,
-                ResolveLibResolver(index_urls, args.timeout, args.cache_dir, state),
-                state,
-            )
+        elif args.project_path is not None:
+            # Determine which kind of project file exists in the project path
+            _source = _create_project_source(args.project_path, state)
+            if _source is None:
+                parser.error(f"couldn't find project file at path: {args.project_path}")
+            source = _source
```

Comment:
IMO it's okay to reduce this a bit -- we can have `_create_project_source` exit the program instead of returning `None` as a signal. I'll take care of that while shepherding this.